saleor/search/backends/base.py

```diff
@@ -101,12 +107,14 @@ class BaseSearchResults(object):
     This class wraps search results. Core search communication is performed
     here.
     """
-    def __init__(self, backend, query, prefetch_related=None):
+
+    def __init__(self, backend, query, prefetch_related=None,
+                 page=None, page_size=None):
         self.backend = backend
         self.query = query
         self.prefetch_related = prefetch_related
-        self.start = 0
-        self.stop = None
+        self.start = (page - 1) * page_size
```

Comment:
Will throw `TypeError` if `page` won't be provided. Either make `page` arg required or set it by default to `int` instead of `None`